service/pipelines/pipelines.go

```diff
@@ -22,74 +22,78 @@ import (
 	"go.opentelemetry.io/collector/processor"
 	"go.opentelemetry.io/collector/receiver"
 	"go.opentelemetry.io/collector/service/internal/capabilityconsumer"
-	"go.opentelemetry.io/collector/service/pipelines"
 )
 
-// Settings holds configuration for building builtPipelines.
+// Settings holds configuration for building a Graph.
 type Settings struct {
+	// Telemetry specifies the telemetry settings.
 	Telemetry component.TelemetrySettings
+	// BuildInfo provides collector start information.
 	BuildInfo component.BuildInfo
 
-	ReceiverBuilder  *receiver.Builder
+	// ReceiverBuilder is a helper struct that given a set of Configs and Factories helps with creating receivers.
+	ReceiverBuilder *receiver.Builder
+	// ProcessorBuilder is a helper struct that given a set of Configs and Factories helps with creating processors.
 	ProcessorBuilder *processor.Builder
-	ExporterBuilder  *exporter.Builder
+	// ExporterBuilder is a helper struct that given a set of Configs and Factories helps with creating exporters.
+	ExporterBuilder *exporter.Builder
+	// ConnectorBuilder is a helper struct that given a set of Configs and Factories helps with creating connectors.
 	ConnectorBuilder *connector.Builder
-
-	// PipelineConfigs is a map of component.ID to PipelineConfig.
-	PipelineConfigs pipelines.Config
 }
 
-type Graph struct {
+// Pipelines of collector components.
+type Pipelines struct {
 	// All component instances represented as nodes, with directed edges indicating data flow.
 	componentGraph *simple.DirectedGraph
 
 	// Keep track of how nodes relate to pipelines, so we can declare edges in the graph.
 	pipelines map[component.ID]*pipelineNodes
 }
 
-func Build(ctx context.Context, set Settings) (*Graph, error) {
-	pipelines := &Graph{
+// New builds a collector pipeline set.
+func New(ctx context.Context, set Settings, cfg Config) (*Pipelines, error) {
```

Comment:
I was mildly confused by the PR because the `Config` type is already in the new package (and it's not a struct, like most Config types are in this code base). Then, there is now a `Settings` alongside `Config` and in the context of this PR it's no longer clear why we have both a settings and a config type. Can you explain?

Reply:
> it's no longer clear why we have both a settings and a config type

In the opentelemetry-collector codebase, `Config` types specify configuration that would be set by the user in the Collector YAML configuration, while `Settings` specifies configuration that would be set only in the context of using the Go API directly.

This change was applied because of @bogdandrutu's suggestion here https://github.com/open-telemetry/opentelemetry-collector/pull/8111#discussion_r1272360657 to make it in line with the extensions.

>  it's not a struct, like most Config types are in this code base

I think the important point of `Config` types is that they can be unmarshaled; I am not sure if there are other examples of configuration not being a struct but in my mind this is not a problem.